pure3.md

```diff
@@ -0,0 +1,225 @@
+
+### Modifications to the Repositories and Vehicle Interactions
+
+In this document, we propose updating one of Uptane's two repositories to store
+in-toto metadata alongside the images. The Image repository is recommended as
+its contents are typically signed for using offline keys, while the Director
+repository usually uses online keys.
+
+When an image is submitted to the Image repository, it must be accompanied by
+its in-toto metadata. The in-toto layout and link metadata for every image are
+noted in the opaque "custom" field of the image's entry in Targets metadata.
+This way, there is no ambiguity about the in-toto metadata to be used when an
+image is fetched and verified. Further, as it may not be practical for the top
+level Targets role to issue new metadata for every update, Uptane's delegations
+model can be used to make this entire process more scalable. Instead, delegated
+roles can be used to sign for the image itself and its in-toto link metadata.
+However, in-toto layouts and the key artifacts used to verify them should be
+signed for by the top level Targets role.
+
+Once the Director repository decides which images are required and signs Targets
+metadata for them, the vehicle must fetch the images and all the metadata using
+the interfaces provided by both the repositories. In this phase, the Director
+repository can continue signing targets metadata for the images, as per the
+default functioning of Uptane, and serve the metadata to the automobile. The
+automobile fetches the image itself from the Image repository along with its
+corresponding in-toto metadata.
+
+### Full Verification ECUs
+
+The Primary ECU receives images and metadata from the repositories, and sends
+them on to the other ECUs that it updates. Each Full Verification ECU is
+extended to perform independent in-toto verification that will be secure even
+if the Primary ECU is compromised. Some Full Verification ECUs, for example the
+Primary, are also expected to perform in-toto verification on behalf of Partial
+Verification ECUs that are resource constrained.
+
+For every target image the client needs to install, the corresponding in-toto
+layout, links, and keys are identified via the custom field of the existing
+Uptane metadata. These files are available as targets on the Image repository.
+Therefore, for each installable target, the updater will need to download and
+verify the Uptane metadata that signs for the in-toto metadata, either the
+Targets role or a delegated role. If verification passes, the actual in-toto
+metadata files—layout, links, and public keys—must be downloaded and their
+hashes must be compared with the Uptane metadata. With these two checks,
+Uptane’s secure delivery properties are extended to the in-toto files. These
+files can then be used with confidence by the client to perform in-toto
+verification.
+
+### Resource Constrained ECUs
+
+The addition of in-toto metadata increases the amount of overhead, which can be
```

Comment:
This extended model neglects the _bandwidth_ overhead imposed by Scudo _within_ a vehicle.

Reply:
I took a pass at updating the document to address this concern. @mnm678 can you also take a look at the resource constraints section now?

Reply:
https://github.com/uptane/pures/pull/9/files#diff-dc35087b2004ea6c7b18c597e94ea4ac00627b24fb9ecda73d2aff12054173cdR146